### PR TITLE
[WIP] Flex plugin update to version 4

### PR DESCRIPTION
**Pull request by GPaoloni**
This PR is an attempt to upgrade the version of flex plugin from v3 to v4. The current state is that it launches the local development server but the test suits are failing (running `npm run test` outputs most of the test suits failing).

Steps done so far:
- Followed https://www.twilio.com/docs/flex/developer/plugins/migrate
- Issues encountered in above guide so continued with https://github.com/twilio/flex-plugin-builder/blob/v4-beta/changelog/v4.md#breaking-changes
- Created `jest.config`, `webpack.config` and `webpack.dev` files (later ones copied from a fresh Felx Plugin created with `twilio-cli`)
- Changed `test` script in plugin.json to directly run jest, as running `flex-plugins test` throws following error
Screenshot from 2021-02-04 16-15-49
- Added `ts-jest` in order to be able to run TS test files (some of them at least).
- Added core babel plugins and `babel.config` file (following several posts).

I believe the remaining issues are arising because of some transpilation steps missing in the jest environment.

### CODE REVIEW

File: plugin-hrm-form/jest.config.js [hunk 1]
```diff
@@ -1,5 +1,9 @@
 module.exports = {
```
Comment (ktalebian) [inline plugin-hrm-form/jest.config.js:1]: @GPaoloni my name is Kousha and I'm the lead-engineer on Flex Plugins. The issue is this configuration file. You are completely overriding the `jest.config.js` that plugin-builder sets this way. What you instead want is:

```js
module.exports = (config, { isProd, isDev, isTest }) => {
  config.transformIgnorePatterns = ['/node_modules/(?!wavesurfer.js)'];
  
  return config;
}
```

Let me know if that works for you on this PR and I can take a look. In the future, feel welcome to just open a PR and tag me on it.

File: plugin-hrm-form/package.json [hunk 2]
```diff
@@ -3,15 +3,14 @@
...
-    "prestart": "npm run bootstrap",
+    "postinstall": "flex-plugin pre-script-check",
+    "prestart": "flex-plugin pre-start-check",
     "dev": "SKIP_PREFLIGHT_CHECK=true NO_MONITORING=true npm start",
     "start": "flex-plugin start",
     "build": "flex-plugin build",
     "predeploy": "npm run build",
     "deploy": "flex-plugin deploy",
-    "test": "flex-plugin test --env=jest-environment-jsdom-sixteen",
+    "test": "jest --env=jest-environment-jsdom-sixteen",
```
Comment (ktalebian) [inline plugin-hrm-form/package.json:13]: Remove this as well, and use `UNBUNDLED_REACT=true twilio flex:plugins:test` instead

File: plugin-hrm-form/jest.config.js (hunk 1 above)
Comment (andvirga) [reply plugin-hrm-form/jest.config.js]: @ktalebian HI! I'm Andres and I belong to the same team as Gian. We've tried your suggestions, however, we're getting new errors when trying to import anything from `@twilio/flex-ui` in our tests. All the errors are related to `serialize.cjs.js` library, here is more info:

```bash
src/___tests__/services/ContactService.test.js
  ● Test suite failed to run

    TypeError: Cannot read property 'length' of undefined

       5 | import { useFormContext, RegisterOptions } from 'react-hook-form';
       6 | import { get, pick } from 'lodash';
    >  7 | import { Template } from '@twilio/flex-ui';
         | ^
       8 | 
       9 | import {
      10 |   Box,

      at Object.serializeStyles (node_modules/@emotion/serialize/dist/serialize.cjs.js:147:12)
      at css (node_modules/@emotion/css/dist/css.cjs.dev.js:12:20)
      at Object.keyframes (node_modules/@emotion/core/dist/core.cjs.dev.js:163:44)
      at Object.<anonymous> (node_modules/react-select/dist/index-aee072ef.cjs.dev.js:986:33)
      at Object.<anonymous> (node_modules/react-select/dist/react-select.cjs.dev.js:24:15)
      at Object.<anonymous> (node_modules/react-select/dist/react-select.cjs.js:6:20)
      at Object.<anonymous> (node_modules/@twilio/flex-ui/bundle/twilio-flex.prod.js:137:773510)
      at n (node_modules/@twilio/flex-ui/bundle/twilio-flex.prod.js:137:355)
      at Object.<anonymous> (node_modules/@twilio/flex-ui/bundle/twilio-flex.prod.js:137:50094)
      at n (node_modules/@twilio/flex-ui/bundle/twilio-flex.prod.js:137:355)
      at Object.<anonymous> (node_modules/@twilio/flex-ui/bundle/twilio-flex.prod.js:137:225754)
      at n (node_modules/@twilio/flex-ui/bundle/twilio-flex.prod.js:137:355)
      at Object.<anonymous> (node_modules/@twilio/flex-ui/bundle/twilio-flex.prod.js:137:441585)
      at n (node_modules/@twilio/flex-ui/bundle/twilio-flex.prod.js:137:355)
      at Object.<anonymous> (node_modules/@twilio/flex-ui/bundle/twilio-flex.prod.js:137:128677)
      at n (node_modules/@twilio/flex-ui/bundle/twilio-flex.prod.js:137:355)
      at Object.<anonymous> (node_modules/@twilio/flex-ui/bundle/twilio-flex.prod.js:137:187622)
      at n (node_modules/@twilio/flex-ui/bundle/twilio-flex.prod.js:137:355)
      at Object.<anonymous> (node_modules/@twilio/flex-ui/bundle/twilio-flex.prod.js:137:186496)
      at n (node_modules/@twilio/flex-ui/bundle/twilio-flex.prod.js:137:355)
      at Object.<anonymous> (node_modules/@twilio/flex-ui/bundle/twilio-flex.prod.js:137:521845)
      at n (node_modules/@twilio/flex-ui/bundle/twilio-flex.prod.js:137:355)
      at node_modules/@twilio/flex-ui/bundle/twilio-flex.prod.js:137:1162
      at node_modules/@twilio/flex-ui/bundle/twilio-flex.prod.js:137:1173
      at node_modules/@twilio/flex-ui/bundle/twilio-flex.prod.js:137:81
      at Object.<anonymous> (node_modules/@twilio/flex-ui/bundle/twilio-flex.prod.js:137:220)
      at Object.<anonymous> (src/components/common/forms/formGenerators.tsx:7:1)
      at Object.<anonymous> (src/states/contacts/reducer.ts:11:1)
      at Object.<anonymous> (src/services/ContactService.ts:4:1)
      at Object.<anonymous> (src/___tests__/services/ContactService.test.js:5:1)

```
Any help would be really appreciated! Thanks in advance!

Comment (ktalebian) [reply plugin-hrm-form/jest.config.js]: @andvirga nice to meet you as well.

Okay I found where the issue is happening, though I am not sure what exactly _is_ the problem. Until I figure this out and do a fix, here is a temp solution:

1) Bump your flex-ui version to 1.23 (i.e. change `"@twilio/flex-uo": "1.23"` - that is remove `^` and use `1.23`)
2) Add `"rxjs-compat": "6.6.3"` to your package.json a well.

That should fix your issues.

**Review by andvirga — APPROVED** [review]
Tested application and run tests, everything is working OK.

Since @GPaoloni started first this PR and then I took it I would like to explain some changes that I've added:

**1. Solved several linting issues**
The new `flex-plugins` library includes some new linting libraries (such as `sonarjs`) that triggered a lot of lint issues. I've solved most of them, ignore others (we can tackle that in future refactors) and disable one particular rule that originated hundreds of issues: `sonarjs/no-duplicate-string`.

**2. Modified YML (CI)**
Change to the new flex-plugins required some changes in our CI process.
The most significant change is that we now need to include TWILIO_ACCOUNT_SID and
TWILIO_AUTH_TOKEN in some npm commands, since these are sensitive values we've stored them under repo secrets with the following names:
- TL_STAGING_ACCOUNT_SID
- TL_STAGING_AUTH_TOKEN

**3.Version downgrade**
The version of `@twilio/flex-ui` was downgraded to 1.23 due to some errors that we were getting in the latest version.

(TL stands for TorontoLine)
These values will remain secret and won't be printed in the console logs.

**Review by nickhurlburt — APPROVED** [review]
I haven't run this myself, but the code makes sense.  I'm fine with moving forward with this whenever you all are ready.

**Pull request merged**